### ISSUE TITLE
.available() method succeeds on iPad ( and shouldn't ? )

### ISSUE THREAD

**Issue by stickbyatlas** [Telerik-Verified-Plugins/HealthKit#21]
Hi there,

Thanks so much for this plugin, which has saved me countless hours of frustration and helplessness.

I'm currently working on a universal app, primarily iPhone, but optimized for iPad as well. The app pulls data from HealthKit where available, and falls back to a manual entry option when it's not present (e.g. iOS 7).

While researching how to run an app with the HealthKit entitlement on a device that doesn't have HK, I came across [this solution] (http://stackoverflow.com/a/25658268/697061). Under the Project Info in Xcode, I removed 'healthkit' from the list of 'Required device capabilities' in the Target Properties list. Then, in the General tab > Linked Frameworks and Libraries, I set 'HealthKit.framework' to Optional instead of Required.

So, when I review 'Capabilities' > HealthKit, I see an alert prompting me to 'Add the "HealthKit" key to your info plist file'. However the app nows run on these devices.

Which brings me to my issue: testing on an iPad 2 running iOS 8.2, I noticed that the .available() method in the HealthKit plugin is succeeding, even though HealthKit isn't currently available on iPads. So, my users will see HealthKit-related buttons that attempt to execute queries against HealthKit, because the app assumes HealthKit is available. These queries fail silently in the debugger.

My current code:

```javascript
		if("device" in window){

			if(window.device.platform == "iOS"){

				if(window.device.version.split(".")[0] == "8"){

					window.plugins.healthkit.available( // succeeds where it shouldn't
						function(isAvailable) {
							controller.set("healthKit", true);
						}
					);

				}
			}
		}
```
I've tested this on an iPad 2 and 4th-gen iPad, both running iOS 8.2. If this isn't a bug in your code (i.e. Apple's fault), I could always filter out iPads when testing the device capabilities. However, this would mean I would need to issue a subsequent app update when Apple makes HealthKit more widely available.

Can you please advise whether this is a bug or not? If it's not, will you take any action in your codebase to pre-empt Apple's upstream bug? For example, if you ran a check like ( `os-version < 8.2 && device-type == "iPad"`), you could return a failure for this and previous OS versions, but leave the door open for a fix in subsequent software updates by Apple. This would future-proof your code, and would save developers from having to issue app updates once Apple issues a patch.

Interested to hear your thoughts, and thanks again!

**Comment by EddyVerbruggen** [comment]
Hey @stickbyatlas thanks for the elaborate issue report.

You code sample reveals the issue: You need to check the value of `isAvailable` in the callback function of `window.plugins.healthkit.available` : it's a boolean indicating whether or not HK is available on the device.

I will look into the 'entitlement' related question as well.

Good luck,
Eddy

**Comment by EddyVerbruggen** [comment]
Hi, I just removed the `.plist` value from the plugin so this should be iPad compatible out of the box from now on. Thanks for your research on this! Fixed in version 0.3.2.

Note for users of older version of the plugin: iPad usage was only broken on 0.3.1 (not before) because that version added the `.plist` bit.

**Comment by stickbyatlas** [comment]
Whoops - I had assumed .available took two function params, where the first was equivalent to onSuccess and the second onError. Thanks for clearing that up so quickly!
